### PR TITLE
refactored batch-wise feature extraction and added explanation to README and the docs

### PR DESCRIPTION
**Pull request by LukasMut**
- added a public mini-batch extraction method alongside the private mini-batch extraction method. Whereas the former can be used with custom dataset and data loader classes outside of the main loop, the latter is used in the main feature extraction method (as done previously).
- added an explanation about how to call and use the public batch extraction method to `README.md` and the docs
- other small refactorings (e.g., output types are now optional args because tensorflow extractors don't use that argument)

### CODE REVIEW

**Review by lciernik — COMMENTED** [review]
There are two automatic warnings left; otherwise, LGTM.

File: thingsvision/core/extraction/torch.py [hunk 1]
```diff
@@ -91,6 +67,26 @@ def register_hook(self, module_name: str) -> None:
...
+        TensorType["b", "d"],
+    ]:
+        self._module_and_output_check(module_name, output_type)
+        self.register_hook(module_name=module_name)
+        act = self._extract_batch(batch, module_name, flatten_acts)
+        if output_type == "ndarray":
+            act = self._to_numpy(act)
+        self._unregister_hook()
+        return act
+
```
Comment (jonasd4) [inline thingsvision/core/extraction/torch.py:89]: @LukasMut did you measure how much overhead does self.register_hook(module_name=module_name) and self._unregister_hook makes?

Comment (jonasd4) [reply thingsvision/core/extraction/torch.py:89]: Also `self._module_and_output_check(module_name, output_type)`

Comment (jonasd4) [reply thingsvision/core/extraction/torch.py:89]: I tested the overhead based on the extraction of a `resnet50` and there is basically no overhead. For 100 iterations with batch size 64 it took and 34.39s without the register and checks and 34.78s with the new implementation.

**Review by jonasd4 — APPROVED** [review]

File: thingsvision/core/extraction/torch.py (hunk 1 above)
Comment (LukasMut) [reply thingsvision/core/extraction/torch.py:89]: Thanks for checking. I will leave it as is then.

File: thingsvision/core/extraction/base.py [hunk 2]
```diff
@@ -91,12 +126,23 @@ def _extract_batch(
...
+            output_type in self.get_output_types()
+        ), f"\nData type of output feature matrix must be set to one of the following available data types: {self.get_output_types()}\n"
+
     def extract_features(
         self,
         batches: Iterator[Union[TensorType["b", "c", "h", "w"], Array]],
         module_name: str,
         flatten_acts: bool,
-        output_type: str = "ndarray",
+        output_type: Optional[str] = "ndarray",
```
Comment (a1247418) [inline thingsvision/core/extraction/base.py:145]: Why is this optional? Is None valid?

**Review by a1247418 — APPROVED** [review]
Looks good!

**Pull request merged**